tests/ibc-hooks/xcs_cw20_test.go

```diff
@@ -0,0 +1,179 @@
+
+	// Create a pool for that token
+	suite.setupCW20PoolAndRoutes(chainA, swaprouterAddr, cw20IbcDenom, amount)
+
+	// Check that the receiver doesn't have any stake
+	stakeAB := suite.GetIBCDenom(ChainA, ChainB, "stake") // IBC denom for stake in B
+	balanceStakeReceiver := osmosisAppB.BankKeeper.GetBalance(suite.chainB.GetContext(), suite.chainB.SenderAccount.GetAddress(), stakeAB)
+	suite.Require().Equal(int64(0), balanceStakeReceiver.Amount.Int64())
+
+	// Transfer the tokens with a memo for XCS
+	swapMsg := fmt.Sprintf(`{"osmosis_swap":{"output_denom":"stake","slippage":{"twap": {"window_seconds": 1, "slippage_percentage":"20"}},"receiver":"chainB/%s", "on_failed_delivery": "do_nothing", "next_memo":{}}}`,
+		chainB.SenderAccount.GetAddress(),
+	)
+	xcsMsg := fmt.Sprintf(`{"wasm": {"contract": "%s", "msg": %s } }`, crosschainAddr, swapMsg)
+	serializedMemo, _ := json.Marshal(xcsMsg)
+	result, ack := suite.TransferCW20Tokens(path, cw20Addr, cw20ics20Addr, crosschainAddr, "100", string(serializedMemo))
+	suite.Require().Contains(string(ack), "result")
+
+	// Relay the packet created by the XCS contract back to the receiver
+	packet, err := ibctesting.ParsePacketFromEvents(result.GetEvents())
+	suite.Require().NoError(err)
+	suite.RelayPacket(packet, AtoB)
+
+	balanceStakeReceiver = osmosisAppB.BankKeeper.GetBalance(suite.chainB.GetContext(), suite.chainB.SenderAccount.GetAddress(), stakeAB)
+	suite.Require().Greater(balanceStakeReceiver.Amount.Int64(), int64(0))
+
+	// Now swap on the other direction
+	receiver2 := chainB.SenderAccounts[1].SenderAccount.GetAddress() // Using a different receiver that has no cw20s
+	cw20Balance := suite.getCW20Balance(chainB, cw20Addr, receiver2)
+
+	swapMsg = fmt.Sprintf(`{"osmosis_swap":{"output_denom":"%s","slippage":{"twap": {"window_seconds": 1, "slippage_percentage":"20"}},"receiver":"chainB-cw20/%s", "on_failed_delivery": "do_nothing", "next_memo":{}}}`,
+		cw20IbcDenom,
+		receiver2,
+	)
+	xcsMsg = fmt.Sprintf(`{"wasm": {"contract": "%s", "msg": %s } }`, crosschainAddr, swapMsg)
+
+	transferMsg := NewMsgTransfer(sdk.NewCoin(stakeAB, sdk.NewInt(10)), suite.chainB.SenderAccount.GetAddress().String(), crosschainAddr.String(), suite.pathAB.EndpointB.ChannelID, xcsMsg)
+	_, recvResult, _, _ := suite.FullSend(transferMsg, BtoA)
+
+	packet, err = ibctesting.ParsePacketFromEvents(recvResult.GetEvents())
+	suite.Require().NoError(err)
+	suite.RelayPacket(packet, AtoCW20)
+
+	// Check that the receiver has 10 less ibc'd stake than before
+	balanceStakeReceiver2 := osmosisAppB.BankKeeper.GetBalance(suite.chainB.GetContext(), suite.chainB.SenderAccount.GetAddress(), stakeAB)
+	suite.Require().Equal(int64(10), balanceStakeReceiver.Amount.Sub(balanceStakeReceiver2.Amount).Int64())
+
+	// Check that the receiver has more cw20 tokens than before
+	newCw20Balance := suite.getCW20Balance(chainB, cw20Addr, receiver2)
+	suite.Require().Greater(newCw20Balance.Int64(), cw20Balance.Int64())
```

Comment:
Why can't we determine the exact amount expected here?

Reply:
We can do it, but since this depends on the underlying pool, any future changes to how pools work or the params we're using for the tests would make this fail. So I figure this was safer. Happy to change it if you think that's better